src/Type/Php/ArrayReplaceFunctionReturnTypeExtension.php

```diff
@@ -23,54 +32,107 @@ public function isFunctionSupported(FunctionReflection $functionReflection): boo
 
 	public function getTypeFromFunctionCall(FunctionReflection $functionReflection, FuncCall $functionCall, Scope $scope): ?Type
 	{
-		$arrayTypes = $this->collectArrayTypes($functionCall, $scope);
+		$args = $functionCall->getArgs();
```

Comment:
is `isFirstClassCallable()` check needed first? The `getArgs()` got assert notice when there is `...` variadic place holder.

https://github.com/nikic/PHP-Parser/blob/20b0d55f6655ef2d73d841f6b7d87117a90f9ee2/lib/PhpParser/Node/Expr/CallLike.php#L32

Reply:
The extensions are not called with first-class callables at all.